Kernel/CMakeLists.txt

```diff
@@ -518,6 +518,8 @@ endmacro()
 if ("${SERENITY_ARCH}" STREQUAL "x86_64")
     add_compile_options(-mcmodel=large -mno-red-zone)
     set_new_alignment(8)
+elseif ("${SERENITY_ARCH}" STREQUAL "aarch64")
+    set_new_alignment(8)
```

Comment:
That's definitely better than passing `-fnew-alignment=4`, but I wonder why we need to pass this at all. Shouldn't the default be correct already? (But it's fine as-is too for now since it's a clear improvement.)

(@BertalanD since you changed new alignment stuff in clang)

Reply:
Passing `-fnew-alignment=4` is definitely bad, because it causes the following KUBSAN error (previously a crash since I ran without KUBSAN):
```
[Kernel]: KUBSAN: addition of unsigned offset to 0x00000000007eff78 overflowed to 0xeeeeeeeeef6dee38
[Kernel]: KUBSAN: at /home/timon/dev/extra-disk-space/serenity/./Kernel/Heap/Heap.h, line 112, column: 9
[Kernel]: Kernel + 0x00000000000ed560  print_location +0x140
[Kernel]: Kernel + 0x00000000000ef5ec  __ubsan_handle_pointer_overflow +0x9c
[Kernel]: Kernel + 0x00000000001a4768  Kernel::Heap<64ul, 187u, 170u>::deallocate(void*) +0x2a8
[Kernel]: Kernel + 0x00000000001a4e40  KmallocGlobalData::deallocate(void*, unsigned long) +0x340
[Kernel]: Kernel + 0x000000000019ee50  kfree_sized(void*, unsigned long) [clone .localalias] +0x9c
[Kernel]: Kernel + 0x00000000000e82d0  Kernel::register_generic_interrupt_handler(unsigned char, Kernel::GenericInterruptHandler&) +0x380
[Kernel]: Kernel + 0x00000000001980ac  Kernel::GenericInterruptHandler::register_interrupt_handler() +0xc8
[Kernel]: Kernel + 0x0000000000198e60  Kernel::IRQHandler::enable_irq() +0xc0
[Kernel]: Kernel + 0x00000000000e3368  init +0x278
[Kernel]: UB is configured to be deadly, halting the system.
```

I agree that using the default is better, so I changed the logic, such that we only set the `new_alignment` for `i686` and `x86`.

Reply:
> I wonder why we need to pass this at all

I originally added the `-fnew-alignment` flag to allow allocating the 16-byte aligned `FPUState` struct with regular `operator new` instead of `kmalloc_aligned` (https://github.com/SerenityOS/serenity/pull/8772/commits/252984360e9c8ee2748fce888b580644e73fad83).

The flag needed because GCC/Clang assume by default that allocator functions will return 16-byte aligned memory. This is not the case with `kmalloc`, which  only guarantees (or at least guaranteed at the time) a 4-byte alignment on i686 and an 8-byte alignment on x86-64. UBSAN complained about unmet alignment guarantees, so I lowered the default alignment to force the compiler to generate `operator new(std::size_t, std::align_val_t)` calls for overaligned objects. (https://github.com/SerenityOS/serenity/pull/8772/commits/73a1fab2ff9592075576d5d11d030e01a9037f0c).

Reply:
Discussed on Discord: For now it's fine to stick with the default alignment (not passing the flag), once this also becomes a problem on aarch64, we can still add the flag at that time.